src/components/TextNodeView.tsx

```diff
@@ -0,0 +1,75 @@
+import { Node } from "prosemirror-model";
+import { Decoration, DecorationSet, EditorView } from "prosemirror-view";
+import { Component } from "react";
+import { findDOMNode } from "react-dom";
+
+import { CompositionViewDesc, TextViewDesc, ViewDesc } from "../viewdesc.js";
+
+import { wrapInDeco } from "./ChildNodeViews.js";
+
+type Props = {
+  view: EditorView | null;
+  node: Node;
+  pos: number;
+  siblingDescriptors: ViewDesc[];
+  decorations: readonly Decoration[];
+};
+
+export class TextNodeView extends Component<Props> {
+  private viewDescRef: null | TextViewDesc = null;
+
+  updateEffect() {
+    const { decorations, siblingDescriptors, node } = this.props;
+    // There simply is no other way to ref a text node
+    // eslint-disable-next-line react/no-find-dom-node
+    const dom = findDOMNode(this);
```

Comment:
I am getting the following warning in the console.  Im thinking this is due to how it interacts with ProseMirror under the hood?

`Warning: findDOMNode is deprecated in StrictMode. findDOMNode was passed an instance of TextNodeView which is inside StrictMode. Instead, add a ref directly to the element you want to reference.`

Reply:
Hi @sethfair! Until React introduces a mechanism for obtaining a ref of a text node, we're stuck using a deprecated function, `findDOMNode`. This will not introduce any issues, but it does unfortunately result in this warning being printed; you can read more about the deprecation here: https://react.dev/reference/react-dom/findDOMNode#alternatives. Note that the authors don't seem to plan on removing `findDOMNode` until they've provided alternatives for all of its use cases, so I feel comfortable using it for this scenario, which has no alternative.